issue_processing/issue_processing.py

```diff
@@ -98,26 +129,133 @@ def reformat(issue_data):
 
     # re-process all issues
     for issue in issue_data:
+
+        # empty container for issue types
+        issue["type"] = []
+
+        # empty container for issue resolutions
+        issue["resolution"] = []
+
+        # if an issue has no eventsList an empty Lists gets created
+        if issue["eventsList"] is None:
+            issue["eventsList"] = []
+
+        # if an issue has no commentsList an empty Lists gets created
+        if issue["commentsList"] is None:
+            issue["commentsList"] = []
+
+        # if an issue has no relatedCommits an empty Lists gets created
+        if issue["relatedCommits"] is None:
+            issue["relatedCommits"] = []
+
+        # if an issue has no relatedIssues an empty Lists gets created
+        if "relatedIssues" not in issue:
+            issue["relatedIssues"] = []
+
+        # add "closed_at" information if not present yet
+        if issue["closed_at"] is None:
+            issue["closed_at"] = ""
+
+        # parses the creation time in the correct format
+        issue["created_at"] = format_time(issue["created_at"])
+
+        # parses the close time in the correct format
+        issue["closed_at"] = format_time(issue["closed_at"])
+
+        # checks if the issue is a pull-request or a normal issue and adapts the type
+        if issue["isPullRequest"]:
+            issue["type"].append("pull request")
+        else:
+            issue["type"].append("issue")
+
+    return issue_data
+
+
+def merge_issue_events(issue_data):
+    """
+    All issue events are merged together in the eventsList. This simplifies processing in later steps.
+
+    :param issue_data: the issue data from which the events shall be merged
+    :return: the issue data with merged eventsList
+    """
+
+    log.info("Merge issue events ...")
+
+    for issue in issue_data:
+
         # temporary container for references
         comments = dict()
 
-        # initialize event
+        # adds creation event to eventsList
         created_event = dict()
         created_event["user"] = issue["user"]
         created_event["created_at"] = issue["created_at"]
         created_event["event"] = "created"
+        created_event["event_info_1"] = "open"
+        created_event["event_info_2"] = []
         issue["eventsList"].append(created_event)
-
-        # add event name to comment and add reference target
+        issue["state_new"] = "open"
+
+        # adds events for related issues to eventsList
+        # GH-Wrapper doesn't provide information about time and user yet. Has to be updated if this changes
```

Comment:
Meanwhile, the GH-Wrapper does provide the information about time and user of related issues (it is not merged to master yet, but we already provided some example data on the new format, see here: https://github.com/se-passau/GitHubWrapper/issues/22#issuecomment-411324641
So, please include this (have a look at the most recent example data we have sent you).